build.gradle

```diff
@@ -563,30 +622,38 @@ dependencies {
     testRuntime 'org.seleniumhq.selenium:selenium-firefox-driver'
     testRuntime 'org.seleniumhq.selenium:selenium-safari-driver'
 
-    compile 'org.slf4j:jcl-over-slf4j'
-    compile 'org.slf4j:jul-to-slf4j'
-    compile 'org.slf4j:log4j-over-slf4j'
-    compile 'org.slf4j:slf4j-api'
+    implementation 'org.slf4j:jcl-over-slf4j'  // Apache/Jakarta Commons Logging -> slf4j
+    implementation 'org.slf4j:jul-to-slf4j'  // java.util.logging -> slf4j
+    implementation 'org.slf4j:log4j-over-slf4j'  // log4j1 -> slf4j; cf. log4j-to-slf4j
+    implementation 'org.slf4j:slf4j-api'  // slf4j itself
 
     // enable `import spock.lang.*`
     testCompile 'org.spockframework:spock-core:1.3-groovy-2.4'
     testCompile 'org.spockframework:spock-spring:1.3-groovy-2.4'
 
+    // enable `import org.springframework.orm.hibernate5.*`
+    implementation 'org.springframework:spring-orm'
+
     // may enable '/health' and '/info' endpoints
     compile 'org.springframework.boot:spring-boot-starter-actuator'
 
-    // set up Spring log framework (logback under the hood)
-    compile 'org.springframework.boot:spring-boot-starter-logging'
+    // set up Spring log framework (slf4j under the hood)
+    implementation 'org.springframework.boot:spring-boot-starter-logging'
 
     /*
-     * Prevent 'Unable to start EmbeddedWebApplicationContext
-     * due to missing EmbeddedServletContainerFactory bean.'
+     * Use `provided` per Grails 3's deployment guide, see
+     * http://docs.grails.org/3.3.16/guide/deployment.html
+     *
+     * We don't support application servers other than Tomcat.
+     *
+     * This ticket has a good discussion of `provided` gotchas.
+     * https://github.com/grails/grails-core/issues/10483
      */
-    compile 'org.springframework.boot:spring-boot-starter-web'
+    provided 'org.springframework.boot:spring-boot-starter-tomcat'
```

Comment:
My understanding is that `provided` declares that a dependency should be compiled against, but not included in the war file. All of the dependencies inside this dependency come "for free" when running on Tomcat (see https://tomcat.apache.org/tomcat-8.5-doc/RELEASE-NOTES.txt). Grails recommends using "provided" here if deploying to a different tomcat release than what you compile against, which we do (Spring boot includes tomcat 8.5.43, obnavtest1 uses Tomcat 9.0.16). (see  http://docs.grails.org/3.3.16/guide/deployment.html)

Using provided here prevents us from including jars that will be present on the class path anyway when deployed to a tomcat server, works correctly when using `grails run-app` and even starts up on jetty, because it provides the same APIs. So I'm almost certain this is the correct declaration here.